assertion/function/assertiontree/assertion_node.go

```diff
@@ -73,6 +72,10 @@ type assertionNodeCommon struct {
 	parent          AssertionNode // this should be nil for the root
 	children        []AssertionNode
 	consumeTriggers []*annotation.ConsumeTrigger
+
+	// originalExpr stores the original call expression that prompted the creation of this assertion node
+	originalExpr ast.Expr // this should be nil for the root
```

Comment:
Wonder if it's just better to call it `Expr`, since there are no "fake" expressions here anyways.